test/e2e/storage/drivers/in_tree.go

```diff
@@ -1223,6 +1224,14 @@ func (l *localDriver) PrepareTest(ctx context.Context, f *framework.Framework) *
 	framework.ExpectNoError(err)
 
 	l.hostExec = utils.NewHostExec(f)
+	// It is recommended to mount /tmp with options noexec, nodev, nosuid.
+	// tmpfs on /tmp type tmpfs (rw,nosuid,nodev,noexec,relatime,seclabel,inode64)
+	// This prevents scripts and binaries from being executed from the /tmp directory.
+	// This can cause errors like "Permission denied" when executing files from `/tmp`.
+	// To pass the test that verifies the execution of files on a volume, we remount `/tmp` with the exec option.
+	remountCmd := `mount | awk '{if ($3 == "/tmp") print $0}' | grep -q "noexec" && mount -o remount,exec /tmp || true`
+	err = l.hostExec.IssueCommand(ctx, remountCmd, l.node)
+	framework.ExpectNoError(err)
```

Comment:
Sorry I did not re-check it sooner. It's not great to modify the host, it may be a real cluster.

If it's hard not to do so, then let's skip exec tests for **some** local volumes. There are all local volume "types" we have in CI:

https://github.com/kubernetes/kubernetes/blob/af41aa1d9f7f347c85aba629963941a8ce6a95fd/test/e2e/storage/utils/local.go#L40-L57

Some of them can't support exec because they're on /tmp (all the bind-mounts, directories etc), but some always can, because they get a fresh mount (like LocalVolumeBlockFS, LocalVolumeTmpfs, LocalVolumeGCELocalSSD, please check the others). I think it's enough if we test exec only for those.

The tests framework has a list of all "exceptional" local volume types, currently with only one entry for LocalVolumeBlock tha can support block volume:
https://github.com/kubernetes/kubernetes/blob/af41aa1d9f7f347c85aba629963941a8ce6a95fd/test/e2e/storage/drivers/in_tree.go#L1149-L1158

The defaults are used for all others:
https://github.com/kubernetes/kubernetes/blob/af41aa1d9f7f347c85aba629963941a8ce6a95fd/test/e2e/storage/drivers/in_tree.go#L1140-L1147

So, you can make the default with `CapExec: false` and then add exception for LocalVolumeBlockFS, LocalVolumeTmpfs, LocalVolumeGCELocalSSD with `CapExec: true`, with some smart comment why false there and true here.